### ISSUE TITLE
Media picker TakePhotoAsync throws exception on HTC phones

### ISSUE THREAD

**Issue by arsindelve** [xamarin/Xamarin.Mobile#35]
I have a Xamarin Forms application which is crashing when we attempt to take a photo using the Media Picker. Here is the code we use:

```
var pictureButton = new Button () { 
Text = "Take Picture", 
Command = new Command (async () => {

var device = Resolver.Resolve<IDevice> ();
var mediaPicker = DependencyService.Get<IMediaPicker> ();

if (mediaPicker == null) { 
    mediaPicker = device.MediaPicker;
}
try {
     MediaFile result = await mediaPicker.TakePhotoAsync (
     new CameraMediaStorageOptions { 
    DefaultCamera = CameraDevice.Rear
   });
  var hello = result.Source;
  } catch (Exception ec) {
        var hello = ec;
 }

  })
};
```

The message we get in the caught exception is “   {System.IO.IOException: Couldn't create directory, have you added the WRITE_EXTERNAL_STORAGE permission? at Xamarin.Forms.Labs.Droid.Services.Media.MediaPickerActivity.GetOutputMediaFile (Android.Content.Context context, System.String subdir, System.String name, Boolean isPhoto) [0x00000] in <filename unknown>:0 at Xamarin.Forms.Labs.Droid.Services.Media.MediaPickerActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x00000] in <filename unknown>:0 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 at System.Runtime.CompilerServices.TaskAwaiter`1[Xamarin.Forms.Labs.Services.Media.MediaFile].GetResult () [0x00000] in <filename unknown>:0 at MainPage+<MainPage>c__async2.MoveNext () [0x000cf] in /Users/Michael/Google Drive/Gateway/Source/LinkGreenMobile/LinkGreen/MainPage.cs:71 }    System.IO.IOException”

I have tripled checked that we do in fact have that permission checked in the manifest. 

This behaviour happens on two different HTC phones. It does not happen on the GenyMotion simulator. However, on these same HTC phones, the MonoDroid sample camera app works fine, so I know these phone are capable of taking photos.

**Comment by preitzel** [comment]
Hi .. did you ever manage to find out why your HTC is crashing when accessing the camera? I have the same problem right now where on Samsung and Sony I don't have any problem but on a HTC device it cause a crash.
If you have a way to solve this I will really appreciate how to fix this

Thanks in advance
Per Reitzel
pr@continia.com

**Comment by iforlife** [comment]
I have the same problem :(

>   at Plugin.Media.MediaPickerActivity.GetOutputMediaFile (Android.Content.Context context, System.String subdir, System.String name, System.Boolean isPhoto, System.Boolean saveToAlbum) [0x0008a] in C:\projects\mediaplugin\src\Media.Plugin.Android\MediaPickerActivity.cs:486 
  at Plugin.Media.MediaPickerActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x00183] in C:\projects\mediaplugin\src\Media.Plugin.Android\MediaPickerActivity.cs:148 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/4468/f913a78a/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x00047] in /Users/builder/data/lanes/4468/f913a78a/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:187 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in /Users/builder/data/lanes/4468/f913a78a/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:156 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in /Users/builder/data/lanes/4468/f913a78a/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:128 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in /Users/builder/data/lanes/4468/f913a78a/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/TaskAwaiter.cs:357 
  at Plugin.Media.MediaImplementation+<TakePhotoAsync>d__14.MoveNext () [0x000d0] in C:\projects\mediaplugin\src\Media.Plugin.Android\MediaImplementation.cs:119

**Comment by Cheesebaron** [comment]
YOu should use: https://github.com/jamesmontemagno/MediaPlugin which actually seems like you are doing. You should probably report bugs there instead.

Also provide full stack trace and exception.

**Comment by preitzel** [comment]
I found out that on HTC devices the TakePhotoAsync returned actually before the camera were finish handling the image. When I tried accessing the image stream my program was crashing.

The way I solved it was to add a small wait loop right after the return of TakePhotoAsync asking for the length of the mediastream every 1 sec. (in my case for max 20 seconds) and when the stream was larger than zero I continued working with the image .. with no problem.
